worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java

```diff
@@ -1,273 +1,291 @@
-                Map<String, Tag<?, ?>> entityEntryMap = entityEntry.getValue();
-                ListTag posTag = (ListTag) entityEntryMap.get("pos");
-                CompoundTag nbtTag = (CompoundTag) entityEntryMap.get("nbt");
-                String id = nbtTag.getString("Id");
-                Location location = NBTConversions.toLocation(clipboard, posTag, nbtTag.getListTag("Rotation"));
-                if (!id.isEmpty()) {
-                    BaseEntity state = new BaseEntity(EntityTypes.get(id), nbtTag);
-                    clipboard.createEntity(location, state);
-                }
+
+        // Blocks
+        final List<LinCompoundTag> blocks = parent.getListTag("blocks", LinTagType.compoundTag()).value();
+        for (final LinCompoundTag block : blocks) {
+            int state = block.getTag("state", LinTagType.intTag()).valueAsInt();
+            if (state >= palette.length) {
+                throw new IOException("state index exceeds palette length");
+            }
+            List<LinIntTag> pos = block.getTag("pos", LinTagType.listTag()).asTypeChecked(LinTagType.intTag()).value();
+            if (size.size() != 3) {
+                throw new IOException("Expected 'pos' to contain 3 integers, but got " + size.size());
             }
+            LinCompoundTag nbt = block.findTag("nbt", LinTagType.compoundTag());
+            if (nbt == null) {
+                clipboard.setBlock(pos.get(0).valueAsInt(), pos.get(1).valueAsInt(), pos.get(2).valueAsInt(), palette[state]);
+                continue;
+            }
+            clipboard.setBlock(
+                    pos.get(0).valueAsInt(), pos.get(1).valueAsInt(), pos.get(2).valueAsInt(),
+                    palette[state].toBaseBlock(nbt)
+            );
+        }
+
+        // Entities
+        LinListTag<@org.jetbrains.annotations.NotNull LinCompoundTag> entities = parent.findListTag("entities", LinTagType.compoundTag());
+        if (entities == null) {
+            return clipboard;
+        }
+        for (final LinCompoundTag entity : entities.value()) {
+            final LinCompoundTag nbt = entity.getTag("nbt", LinTagType.compoundTag());
+            clipboard.createEntity(
+                    NBTConversions.toLocation(clipboard,
+                            entity.getListTag("pos", LinTagType.doubleTag()),
+                            nbt.getListTag("Rotation", LinTagType.floatTag())
+                    ),
+                    new BaseEntity(
+                            EntityTypes.get(nbt.getTag("id", LinTagType.stringTag()).value()),
+                            LazyReference.computed(dataFixer.fixUp(DataFixer.FixTypes.ENTITY, nbt))
+                    )
+            );
         }
         return clipboard;
     }
 
     @Override
     public void write(Clipboard clipboard) throws IOException {
         clipboard.flush();
-        write(clipboard, "FAWE");
-    }
-
-    public void write(Clipboard clipboard, String owner) throws IOException {
```

Comment:
Is there a particular reason for removing owner? We should also have it as a deprecated method still to ensure API compatibility

Reply:
> author: Name of the player who created this structure. Only exists for structures saved before 1.13.

https://minecraft.wiki/w/Structure_file


Will add method again and delegate to `write(Clipboard)`